src/compiler/checker/lines_32000_40000.rs

```diff
@@ -179,7 +183,26 @@ impl TypeChecker {
         self.check_property_declaration(node)
     }
 
+    pub(super) fn get_effective_type_arguments(
+        &self,
+        node: &Node, /*TypeReferenceNode | ExpressionWithTypeArguments*/
+        type_parameters: &[Rc<Type /*TypeParameter*/>],
+    ) -> Vec<Rc<Type>> {
+        self.fill_missing_type_arguments(
+            map(
+                Some(node.as_has_type_arguments().maybe_type_arguments().unwrap()),
+                |type_argument, _| self.get_type_from_type_node(type_argument),
+            ),
+            Some(type_parameters),
+        )
+        .unwrap()
+    }
+
     pub(super) fn check_type_reference_node(&mut self, node: &TypeReferenceNode) {
+        maybe_for_each(node.type_arguments.as_ref(), |type_argument, _| {
```

Comment:
I guess I added a `maybe_for_each()` variant of `for_each()` whose argument is an `Option`